### ISSUE TITLE
Using batch normalization in the generator network leads to mode collapse

### ISSUE THREAD

**Issue by yliu120** [martinarjovsky/WassersteinGAN#40]
I tried WGAN with Tensorflow and found that when batch normalization is not used in the generator (MLP network), then there is no mode collapse. However, when I added batch normalization between hidden layers, partial mode collapse occurred. I read your code and found that your MLP_G doesn't use any batch norm layers so I am wondering whether you have tried to add batch normalization in the generator only to see whether they will cause partial mode collapse.

Or, anyone already has some experience/ideas on this?

**Comment by martinarjovsky** [comment]
I haven't tried batch normalization with MLPs sadly. Sorry I can't be of further help.
